src/idl/src/annotation.c

```diff
@@ -133,6 +133,31 @@ annotate_autoid(
   return IDL_RETCODE_OK;
 }
 
+static idl_retcode_t
+annotate_optional(
+  idl_pstate_t *pstate,
+  idl_annotation_appl_t *annotation_appl,
+  idl_node_t *node)
+{
+  if (!idl_is_member(node)
+   && !idl_is_case(node)) {
+        idl_error(pstate, idl_location(annotation_appl),
+          "@optional can only be assigned to member or case nodes");
+        return IDL_RETCODE_SEMANTIC_ERROR;
+  }
+
+  assert(annotation_appl);
+  if (annotation_appl->parameters
+   && annotation_appl->parameters->const_expr
+   && idl_mask(annotation_appl->parameters->const_expr) != (IDL_LITERAL | IDL_BOOL)) {
+        idl_error(pstate, idl_location(annotation_appl),
+          "@optional cannot have non-boolean values assigned");
+    return IDL_RETCODE_SEMANTIC_ERROR;
+  }
+
```

Comment:
Shouldn't `((idl_member_t *)node)->is_optional` be set at this point?

Reply:
This is no longer relevant, after rebasing I have changed to using the IDL_ANNOTATABLE macro